### DISCUSSION TITLE
Disable created wifi AP

### DISCUSSION THREAD

**Discussion by Luke-Shepp** [rsta2/circle#449] [Q&A] [answered]
Hi,

I'm getting this project set up and it's working well for my uses on a Pi Zero 2W. 

I'm using `CBcm4343Device::CreateOpenNet` to have the ability to connect to it from other devices, along with something similar to `21-webserver` to allow configuration changes from a web interface - all works good. 

I'm looking for a way to disable this network (after a button press on a gui using lvgl, but that's fine, just for context), and re-enable it at a later time with CreateOpenNet again. 

I'm very new to C, so I might be missing something obvious!

I've tried;

This is from the destructor just pulled out into something usable
```
void CBcm4343Device::Shutdown (void)
{
	assert (s_EtherDevice.shutdown != 0);
	(*s_EtherDevice.shutdown) (&s_EtherDevice);

	delete s_EtherDevice.oq;
}
```
I get:
![IMG_1231](https://github.com/rsta2/circle/assets/35956374/f42ab8a1-64d1-4ece-94c0-abc6a67fadeb)

I've also tried using the disassoc like:
```
m_WLAN.Control("disassoc 1");
```

But get:
![IMG_1232](https://github.com/rsta2/circle/assets/35956374/f764d089-46c5-455e-806c-9ca2d27596e3)

I've tried to do my due dilligence and searching both for circle and p9 but I'm afraid I can't see anything useful related to this, so hopefully a post here will resolve!

Thanks in advance

**Comment by rsta2** [comment]
Unfortunately this is not implemented yet. I will try to provide a solution 
soon.

**Comment by rsta2** [comment, marked as answer]
There is a new method `CBcm4343Device::DestroyOpenNet()` implemented on the *develop* branch now. You can call it to shutdown a created open network. Later you can re-create the network by calling `CBcm4343Device::CreateOpenNet()` as before. The `CBcm4343Device` object must not be deleted in the meantime. Tested on a RPi Zero 2 W.

**Reply by Luke-Shepp** [reply]
Absolutely perfect - thank you!
